skbio/tree/_tree.py

```diff
@@ -2430,10 +2432,14 @@ def update_result():
             starts, stops = [], []  # to calc ._start and ._stop for curr node
             for child in node.children:
                 if child.length is None:
-                    raise NoLengthError("Node with name '%s' doesn't have a "
-                                        "length." % child.name)
+                    warnings.warn("Node with name '%s' doesn't have a "
+                                  "length." % child.name,
+                                  RepresentationWarning)
 
-                distances[child.__start:child.__stop] += child.length
+                    distances[child.__start:child.__stop] += 0
+
+                else:
+                    distances[child.__start:child.__stop] += child.length
```

Comment:
Minor stylistic comment: there's a small amount of duplication in the if/else branches (`distances[child.__start:child.__stop] += ...`). It'd be cleaner to store the length in a local variable within the if/else and then have a single statement after the if statement that sets the value.